cli/mender-artifact/modify.go

```diff
@@ -200,5 +203,117 @@ func modifyExisting(c *cli.Context, image string) error {
 		}
 	}
 
+	err := modifyArtifactAttributes(c, image)
+	if err != nil {
+		return err
+	}
+
+	err = modifyPayloadAttributes(c, image)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func modifyArtifactAttributes(c *cli.Context, image VPImage) error {
```

Comment:
Needs convincing: I initially dont like the payload vs artifact attribute separation.  Whats the benefit?

Reply:
There is a very clear separation in the spec, and look also how they use very different methods and structs to do their job. Granted, we haven't so far allowed more than one payload per artifact, but we should expect this to happen (at least one customer has asked for this already). `modifyPayloadAttributes` should actually work on an array, not a single instance, as opposed to `modifyArtifactAttributes`.

Reply:
I realize it is in the spec, and that this is coming soon, it was more of an architectural thing, in the code. if `modifyPaylodAttributes` accepted an array it should be named in the plural at least. ( I know it doesn't as of now). But for the future: Is this supposed to work on the parts that have already been extracted through parsing an artifact, and then only sending in the already parsed parts of the update?